### PR TITLE
fix: remove ecms dependency and rework the history version for duplicated file - EXO-61044

### PR DESCRIPTION
**Pull request by GouadriaHanen**
prior to this change, after duplicating a document, the history version drawer is empty since no version is added
after this change, an auto version is added after duplicating a document

### CODE REVIEW

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java [hunk 1]
```diff
@@ -699,10 +729,7 @@ public AbstractNode duplicateDocument(long ownerId, String fileId, String prefix
         newNode = duplicateItem(oldNode, parentNode, parentNode, prefixClone);
         parentNode.save();
       }
-      AutoVersionService autoVersionService = CommonsUtils.getService(AutoVersionService.class);
-      if (autoVersionService != null) {
-        autoVersionService.autoVersion(newNode);
-      }
+      autoVersion(newNode);
```
Comment (ahamdi) [inline documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java]: COuld you use org.exoplatform.services.jcr.ext.utils.VersionHistoryUtils#createVersion to create the new version ?

File: documents-storage-jcr/src/test/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorageTest.java [hunk 2]
```diff
@@ -189,10 +186,8 @@ public void duplicateDocument() throws Exception {
     when(currentNode.addNode("copy of test","nt:file")).thenReturn(currentNode);
     when(identity.getRemoteId()).thenReturn("username");
     when(JCRDocumentsUtil.getUserSessionProvider(repositoryService, userID)).thenReturn(sessionProvider);
-    when(CommonsUtils.getService(AutoVersionService.class)).thenReturn(autoVersionService);
     jcrDocumentFileStorage.duplicateDocument(1L,"1","copy of",userID);
     verify(sessionProvider, times(1)).close();
```
Comment (ahamdi) [inline documents-storage-jcr/src/test/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorageTest.java:192]: we need to asssert that the file is already versioned

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java [hunk 3]
```diff
@@ -110,6 +112,35 @@ public class JCRDocumentFileStorage implements DocumentFileStorage {
...
+      //ignore invalid input config
+      if(maxAllowVersion < 0) maxAllowVersion = DOCUMENT_AUTO_DEFAULT_VERSION_MAX;
+      if(maxLiveTime < 0) maxLiveTime = DOCUMENT_AUTO_DEFAULT_VERSION_EXPIRED;
+    }catch(NumberFormatException nex){
+      maxAllowVersion = DOCUMENT_AUTO_DEFAULT_VERSION_MAX;
+      maxLiveTime = DOCUMENT_AUTO_DEFAULT_VERSION_EXPIRED;
+    }
+    maxLiveTime = maxLiveTime * 24 * 60 * 60 * 1000;
+  }
+
```
Comment (ahamdi) [inline documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java]: this is no more needed right ?

Comment (GouadriaHanen) [reply documents-storage-jcr/src/main/java/org/exoplatform/documents/storage/jcr/JCRDocumentFileStorage.java]: yes removed !

**Review by ahamdi — APPROVED** [review]
LGTM

**Pull request merged**